### PR TITLE
type enums everywhere

### PR DESCRIPTION
**Pull request by density215**
- Expanded the typeenum macro to allow for ranges with data-field variants
- Use the enum instead of manual impl for several types in bmp

### CODE REVIEW

File: src/bmp/message.rs [hunk 1]
```diff
@@ -1105,19 +1097,19 @@ impl<'a> Display for InformationTlv<'a> {
...
+    /// See also
+    /// <https://www.iana.org/assignments/bmp-parameters/bmp-parameters.xhtml#initiation-peer-up-tlvs>
+    InformationTlvType, u16,
+    0 => String,
+    1 => SysDesc,
+    2 => SysName,
+    3 => VrfTableName,
+    4 => AdminLabel;
+    5.. => Undefined
+);
```
Comment (ximon18) [inline src/bmp/message.rs:1112]: Unless we have it elsewhere, keeping undefined values around might be handy. Instead this PR goes the other way and discards a value we previously kept. Not sure about that at first glance.

Comment (ximon18) [reply src/bmp/message.rs:1112]: It also loses the RFC references which might be handy to keep.

Comment (density215) [reply src/bmp/message.rs:1120]: It doesn't discard the value, it passes it in the enum variant (that's not expressed in the macro)

File: src/util/macros.rs [hunk 2]
```diff
@@ -7,56 +7,96 @@
...
+///         3 => LocalRibInstance,
+///     }
+///     {
+///         4..=250 => Unassigned,
+///         251..=254 => Experimental,
+///         255 => Reserved
+///     }
+/// );
+/// ```
+/// Note that match lines with ranges are separated by semi-colons, rather
```
Comment (ximon18) [inline src/util/macros.rs:42]: This comment is outdated now, right?

File: src/bmp/message.rs (hunk 1 above)
Comment (partim) [reply src/bmp/message.rs:1120]: Re the RFC reference: You could extend the macro to also allow per-variant attributes.

**Review by ximon18 — APPROVED** [review]

File: src/util/macros.rs (hunk 2 above)
Comment (density215) [reply src/util/macros.rs]: it is

**Review by DRiKE — APPROVED** [review]

**Pull request merged**